docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx

```diff
@@ -133,54 +130,174 @@ In the following table, we compare the supported methods for managing and passin
               marginTop: "0px",
             }}
           >
-            You're using{" "}
-            <a href="/dagster-cloud/deployment/hybrid">Hybrid deployment</a>
+            Maximum of 1,000 variables per full deployment
+          </li>
+          <li>Variables must be less than or equal to 4KB in size</li>
+          <li>
+            Variable names:
+            <ul>
+              <li>Must be 512 characters or less in length</li>
+              <li>Must start with a letter or underscore</li>
+              <li>Must contain only letters, numbers, and underscores</li>
+              <li>
+                May not be the same as{" "}
+                <a href="#built-in-variables">built-in (system) variables</a>
+              </li>
+            </ul>
           </li>
         </ul>
       </td>
+      <td>
+        Variable names:
+        <ul>
+          <li>Must start with a letter or underscore</li>
+          <li>Must contain only letters, numbers, and underscores</li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <strong>Storage and encryption</strong>
+      </td>
+      <td>
+        Uses Amazon Key Management Services (KMS) and envelope encryption. Refer
+        to the{" "}
+        <a href="#storage-and-encryption">Storage and encryption section</a> for
+        more info.
+      </td>
+      <td>Dependent on provider</td>
+    </tr>
+    <tr>
+      <td>
+        <strong>Scope</strong>
+      </td>
+      <td>
+        Scoped by deployment (full and Branch Deployments) and code location
+      </td>
+      <td>Scoped by code location</td>
     </tr>
   </tbody>
 </table>
 
+### Storage and encryption
+
+<Note>
+  This section is applicable only if using the Dagster Cloud UI to manage
+  environment variables.
+</Note>
+
+To securely store environment variables defined using the Dagster Cloud UI, Dagster Cloud uses [Amazon Key Management Services (KMS)](https://docs.aws.amazon.com/kms/index.html) and [envelope encryption](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping). Envelope encryption is a multi-layered approach to key encryption. Plaintext data is encrypted using a data key, and then the data under the data key is encrypted under another key.
+
+Here's a look at how it works in Dagster Cloud:
+
+<center>
+  <Image
+    alt="Dagster Cloud encryption key hierarchy diagram"
+    src="/images/dagster-cloud/developing-testing/environment-variables/encryption-key-hierarchy.png"
+    width={836}
+    height={561}
+  />
+</center>
+
+In Dagster Cloud, each customer account is assigned a unique key, which then encrypts the data associated with that account. All customer keys are encrypted with a single, non-exportable AWS KMS master key.
+
+This approach isolates each account's data and reduces the risk of exposure by limiting the amount of data a single key can access.
```

Comment:
Out of curiosity, would this still be a nit if each secret under the account-level key were encrypted with its own data key instead of all secrets being encrypted with the account-level key?